Render travel advice parts

steventux opened the pull request:
https://trello.com/c/vE8tzKcn/388-render-multi-part-guides

![screenshot from 2016-01-07 14 31 23](https://cloud.githubusercontent.com/assets/93511/12172869/70803312-b54b-11e5-9ae0-0fa6634ca488.png)

Renders Travel Advice for a country.
This handles the basic part navigation and view rendering for travel advice, it omits breadcrumbs and related links and print pages as these are covered in other stories.
\cc @tuzz @danielroseman @dsingleton

Commit:
Add content store initializer

Commit:
Add simple content item route...

Makes a simple request for a single content item.

Commit:
Add debugging tools

tuzz commented on app/controllers/multipage_contents_controller.rb:
```diff
@@ -0,0 +1,31 @@
+    else
+      render_404
+    end
+  end
```
If the user requests a part, but that part doesn't exist on the content item, would this 404?

steventux replied:
No, the idea (taken from [alphagov/frontend](https://github.com/alphagov/frontend/blob/master/app/controllers/travel_advice_controller.rb#L51..L69) is that unless the current part exists on the content item we are redirected to the base country path. Only if the content item doesn't exist do we 404. So `/foreign-travel-advice/legoland` is a `404` whereas `/foreign-travel-advice/spain/part-does-not-exist` redirects to `/foreign-travel-advice/spain`.
[Tests here](https://github.com/alphagov/multipage-frontend/commit/295456d4eecd95c4d9acad07f2286eea7b92e23c#diff-794b79da3239792939aa3993f60355abR27) cover these 3 scenarios.

tuzz commented on app/models/travel_advice.rb:
```diff
@@ -0,0 +1,35 @@
+
+  DETAILS_ATTRIBUTES = %i(
+    summary change_description alert_status email_signup_link
+  )
```
Rather than metaprogram some attr methods, would it be clearer to define those attrs here on the subclass?

steventux replied:
Yes, I agree, it's an uncomfortable pattern for me too, was trying to escape `OpenStruct` but this ended up being quite ugly in itself. I wonder how much of this could be worked into presenters...

tuzz commented on config/initializers/content_store.rb:
```diff
@@ -0,0 +1,21 @@
+  end
+
+  def self.content_store_name
+    ENV["DRAFT_CONTENT_STORE"] ? "draft-content-store" : "content-store"
```
Could potentially go into application.rb as a Rails.config

tuzz commented:
Looks fine to me. A bit metaprogrammery in parts, but the app's simple and straightforward enough to understand.

Commit:
Add frontend toolkit and tidy up dependencies

Commit:
Add a base multipage content model...

This is the base model for various multiple page formats which contain
content parts particular to each page.

steventux commented:
@tuzz I've reworked this as per our discussion, after further review this should be enough to make a start on print and atom feed stories.

Commit:
Add travel advice country format model

Commit:
Add a base multipage content controller...

Handles the common redirecting and rendering of content divided into
parts.

Commit:
Add travel advice controller

Commit:
Add slimmer template

Commit:
Make public_updated_at a toplevel attribute

Commit:
Assign image and document assets to travel advice

Commit:
Dedupe development and test group

Commit:
Add previous_part and next_part methods

Commit:
Add travel advice views and scss

Commit:
Add travel advice image assets

Commit:
Add Capybara gem

Commit:
country is a mandatory travel advice field

Commit:
Add api url for travel advice country

Commit:
Fix asset_host config

danielroseman commented on app/controllers/multipage_contents_controller.rb:
```diff
@@ -0,0 +1,31 @@
+    "/#{params[:slug]}"
+  end
+
+  def current_part
```
I don't think this belongs on the controller, especially as we then use it to set a model instance variable that is depended on by other model methods such as next_part. Could we maybe move it to the model and pass in part_slug in the constructor, or have an explicit setter for current_part?

steventux replied:
Good point, this started as a controller instance var prior to the addition of `@content.current_part`, will update.

Commit:
Move the assignment of the current part...

As discussed this should rightly be part of the initialisation of the
model as it's an internal field now.

steventux commented:
@danielroseman updated in light of yr comment https://github.com/alphagov/multipage-frontend/commit/9f2669d31e7abe4a98c0eab15058ac105c255d47

Commit:
Use email signup url in the content item

Commit:
Embelish parts navigation in show travel advice feature spec

Commit:
Simplify content store configuration...

Puppet handles the configuration of draft frontends to draft content
store so this doesn't need to be done in the application. Makes content
store a module attribute on MultipageFrontend.

Commit:
Add a consistent handler for response errors

Commit:
Fix current_part rendering bug...

A bug introduced in
https://github.com/alphagov/multipage-frontend/commit/9f2669d31e7abe4a98c0eab15058ac105c255d47
removed the @current_part assignment. This change makes sure there's
test coverage for navigating parts and fixes the bug.
